### DISCUSSION TITLE
Velocity from offset

### DISCUSSION THREAD

**Discussion by riyasmjgeo** [isce-framework/isce2#536] [General]
Hello All,

Greetings!
I wanted to compute the Ice velocity from a pair of Sentinel-1 SLC by the offset tracking method using topsApp in ISCE2. I have performed the offset mapping, and there are two bands in the output files. I believe the first band represents the pixel offset in azimuth and the second band represents the same in range. 
I wonder how to derive the velocity from these products? I can multiply the offset pixels with pixel spacing and divide it by the temporal gap to derive the rate of change. However, instead of a single velocity image, the result will be in different bands (velocity in azimuth direction and velocity in range direction). 
Should I take either of them according to the SAR geometry and flow direction? Or do I need to find a flow vector using them? If anyone knows any material to understand this system, please let me know.
Any notes, descriptions, videos, or any information are welcome.
Kindly help. Thank you.

**Comment by piyushrpt** [comment]
See this for greater detail: https://www.mdpi.com/2072-4292/13/4/749/pdf

**Comment by mintaizu** [comment]
Sincerely, which two images did you use after topsApp.py processing to do the offset tracking? Is it done with Autorift as a tool?

**Comment by EJFielding** [comment]
The `topsApp.py` application has an option to run dense offsets or offset tracking. It uses the coregistered SLC images.

**Reply by mintaizu** [reply]
Sorry to bother you again, I did not generate a jointly coregistered SLC image using topsApp.py. Only two burst images are in SLC format. May I ask where my problem lies? Is the XML file not configured correctly? Or are there other issues? Could you please help me answer again.

**Reply by EJFielding** [reply]
You need to provide more information about what you did before anybody can provide suggestions.

**Reply by mintaizu** [reply]
Ok, I am going to use two Sentinel 1A data for offset tracking. The tool I use is autoRIFT. According to your previous comments and the instruction file of autoRIFT, I know that I need to use topsApp.py to preprocess the sentinel data, generate slc files and then use autoRIFT, but I use topsApp.py and No slc file was generated. So I tried to use the stacksentinel.py script for processing. stacksentinel.py did generate two slc.full files for co-registration, but the .mat file generated by offset tracking of slc.full always shows NAN values, this is confusing me. I'm currently stuck at this step, can you give me some advice? Thank you again!

**Reply by EJFielding** [reply]
I am sorry, I don't know how to prepare the Sentinel-1 data for autoRIFT. I have never used it. 
If you are using the topsStack or stackSentinel workflow, then it may be difficult to use the coregistered SLC images because the stack processor keeps those as separate bursts. It creates only a virtual dataset for the combined full-resolution SLC images to save space. The "20210622.slc.full.vrt  20210622.slc.full.xml" files are only virtual datasets. I don't know if autoRIFT knows how to work with those virtual datasets.

**Reply by mintaizu** [reply]
Okay, thank you.

**Comment by mintaizu** [comment]
Ok, I am going to use two Sentinel 1A data for offset tracking. The tool I use is autoRIFT. According to your previous comments and the instruction file of autoRIFT, I know that I need to use topsApp.py to preprocess the sentinel data, generate slc files and then use autoRIFT, but I use topsApp.py and No slc file was generated. So I tried to use the stacksentinel.py script for processing. stacksentinel.py did generate two slc.full files for co-registration, but the .mat file generated by offset tracking of slc.full always shows NAN values, this is confusing me. I'm currently stuck at this step, can you give me some advice? Thank you again!

**Reply by EJFielding** [reply]
The `topsApp.py` application will generate the full-resolution merged SLC images if you include the denseoffsets option and run it until the last step.
```
    <property name="do denseoffsets">True</property>
```

**Reply by mintaizu** [reply]
Okay, thank you. I will try your suggestion

**Reply by mintaizu** [reply]
I'm sorry to bother you again, but what attributes do I need to add to the XML file to complete the 'mergebursts' step using topsApp.py? Or how should the topsApp.xml file be configured? Could you please help me answer this question? Thank you, teacher!

**Reply by EJFielding** [reply]
You should add the "denseoffsets" property that I mentioned above. The rest of the topsApp.xml file can be the same as for making a basic interferogram. See the `examples/input_files/topsApp.xml` for an example input file.
